pycaret/time_series/forecasting/oop.py

```diff
@@ -5153,13 +5153,17 @@ def _get_y_data(self, split: str = "all", data_type: str = "original") -> pd.Ser
             elif split == "all":
                 y = self.y
         elif data_type == "imputed":
-            y, _ = _get_imputed_data(pipeline=self.pipeline, y=self.y, X=self.X)
+            y, _ = _get_imputed_data(
+                pipeline=self.pipeline_fully_trained, y=self.y, X=self.X
+            )
```

Comment:
should this be moved into the `else` block below? since the `if split == "train"` block overwrites the variable anyway.

Reply:
updated with some additional branches